CONTRIBUTING.rst

```diff
@@ -33,6 +33,34 @@ The process for submitting source code PRs is generally more involved
 If you're planning a larger change, the contributor guides (in the ``guides/``
 directory) will make sure you're on the right track.
 
+----------------------------------
+Installing from source and testing
+----------------------------------
+
+If you want to install directly from the source code (e.g. because you want to
+make changes and install the changed version) you can do this with:
+
+.. code:: bash
+
+  pip install -r requirements/test.in
+  pip install -r requirements/tools.in
```

Comment:
Should we recommend installing from requirements.txt, so they get known tested versions of our dev dependencies?

Reply:
I was thinking that the pinned version is not completely pinned, because there are e.g. requirements on Windows that don't appear there, but pinning most is still better than nothing!

(though the whole pinning thing is fairly misleading when this comes up - for example colorama is not pinned at all)